### PR TITLE
Output schema to metadata

### PR DESCRIPTION
**Pull request by live1206**
- [AB#411064](https://ceapex.visualstudio.com/Engineering/_workitems/edit/411064)
- Return before applying template while output type is `json` for data type

### CODE REVIEW

**Review by yufeih — APPROVED** [review]

**Review by AngryBerryMS — APPROVED** [review]

**Review by herohua — APPROVED** [review]

File: docs/specs/schema.yml [hunk 1]
```diff
@@ -420,8 +420,9 @@ inputs:
   _themes/ContentTemplate/TestData.html.tmpl:
 outputs:
   a.json: |
-    {"metadata": {"href": "b"}}
-  b.json:
+    {"metadata": {"href": "b", "schema": "TestData"}}
```
Comment (yufeih) [inline docs/specs/schema.yml:423]:
```suggestion
    {"metadata": {"href": "b"}, "schema": "TestData"}
```

**Review by yufeih — CHANGES_REQUESTED** [review]

File: src/docfx/build/page/PageBuilder.cs [hunk 2]
```diff
@@ -134,6 +134,7 @@ public void Build(ErrorBuilder errors, FilePath file)
 
             var systemMetadataJObject = JsonUtility.ToJObject(systemMetadata);
 
+            outputModel["schema"] = mime.Value;
```
Comment (yufeih) [inline src/docfx/build/page/PageBuilder.cs:137]: Move this line to line 152 in case this key is overridden by other processes.

Comment (live1206) [reply src/docfx/build/page/PageBuilder.cs:137]: ok

Comment (yufeih) [inline src/docfx/build/page/PageBuilder.cs]: Also consider adding `schema` to each entry in `.publish.json`

Comment (live1206) [reply src/docfx/build/page/PageBuilder.cs]: Where is this requirement from?
Not related to the current feature.

Comment (yufeih) [reply src/docfx/build/page/PageBuilder.cs]: No, this isn't required

**Review by yufeih — APPROVED** [review]

File: src/docfx/build/page/PageBuilder.cs (hunk 2 above)
Comment (live1206) [reply src/docfx/build/page/PageBuilder.cs]: It shows up in `.publish.json` already by [Extensiondata](https://github.com/dotnet/docfx/blob/a3751d30b15eea5565167bb3dfc1ede7691cf268/src/docfx/publish/PublishItem.cs#L36)

**Pull request merged**